### ISSUE TITLE
[BUG] Searching in the `activity_user_id` field on `mail.activity.mixin` includes INACTIVE records, resulting in unintuitive results

### ISSUE THREAD

**Issue by AaronBeaudoin** [odoo/odoo#87098]
## Steps to Reproduce

1. Create a CRM lead and create a scheduled activity assigned to someone.
2. Mark the scheduled activity as done.
3. Create another scheduled activity on the same CRM lead assigned to a **different user**.
4. Back on the CRM lead list view, add a custom filter on the "Responsible User" (`activity_user_id`) field, searching for the name of the user from the **first scheduled activity** (which should now be inactive).
5. You will see the CRM lead we created is included in the results, which is super confusing to users because they expect to be searching for the **current responsible user**. Even the `activity_user_id` field shows the `user_id` from the **active** scheduled activity, so this really is confusing even when searching using the `model.search()` method.

## Current Behavior

Searching the "Responsible User" field searches users from **inactive** scheduled activities. (At least on CRM lead records; I have not tested to see whether this behavior is the same on other records with scheduled activities.)

## Expected Behavior

Searching the "Responsible User" field searches users from only **active** scheduled activities. This is the intuitive behavior that users would expect when searching for CRM leads **based on who they are assigned to**.

## Impacted Versions

I'm on v13, so at least that version.

**Comment by AaronBeaudoin** [comment]
Also, upon further investigation it seems that records can have **multiple** active scheduled activities, so in that case what should the value of the `activity_user_id` field even be? Perhaps the purpose and behavior of the related fields on the `mail.activity.mixin` model needs some rethinking?

**Comment by ray-odoo** [comment]
"who they are assigned to" is defined by the Salesperson (user_id) Field.

If you look at the XML for the ControlPanelView(SearchView):

**Comment by AaronBeaudoin** [comment]
That is true, but it also doesn't change the validity of this issue. If you are searching a field called "Responsible User" it doesn't make sense to include **inactive** records.
